uaclient/status.py

```diff
@@ -290,6 +290,15 @@ class CanEnableFailureReason(enum.Enum):
 FIPS kernel is running in a disabled state.
   To manually remove fips kernel: https://discourse.ubuntu.com/t/20738
 """
+NOTICE_WRONG_FIPS_METAPACKAGE_ON_CLOUD = """\
+Warning: FIPS kernel is not optimized for your specific cloud.
```

Comment:
While I don't really like the multi-step NOTICES messaging here, the alternatives would be:
 1. bake this logic into either reboot_cmds which ultimately would require two reboots to transition out of the generic fips kernel after next reboot 
-  OR -
2. add a background process to run right after postinst which would perform ua disable fips, apt remove ubuntu-fips; ua enable fips steps (minus the reboot) and add a MESSAGE_FIPS_REBOOT_REQUIRED.

Option 2 would be nicer in that we would only tell the customer "FIPS support requires system reboot to complete configuration."

Either of these options feel a bit heavyweight for this bug fix.  But, let's discuss with @mitechie as we consider whether either of these cases are worth the effort or usability concerns.

Reply:
For option 2, we will also need to modify how we remove FIPS reboot notices on the code. Even if we manually add this notice after running those commands, if the user runs `ua status` just after it finishes, we will remove it because of this [check](https://github.com/canonical/ubuntu-advantage-client/blob/main/uaclient/entitlements/fips.py#L186)